src/xlsx.js

```diff
@@ -21,27 +21,28 @@ export class Workbook {
   }
 }
 
-function extract(sheet, {range, headers = false} = {}) {
+function extract(sheet, {range, headers} = {}) {
   let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
-  const headerRow = headers && sheet._rows[r0++];
+  const headerRow = headers ? sheet._rows[r0++] : null;
   let names = new Set(["#"]);
   for (let n = c0; n <= c1; n++) {
-    let name = (headerRow ? valueOf(headerRow._cells[n]) : null) || toColumn(n);
+    let name = (headerRow ? valueOf(headerRow._cells[n]) : null) || toColumn(n); // should a falsey header value be ignored? or just nullish?
     while (names.has(name)) name += "_";
     names.add(name);
   }
   names = new Array(c0).concat(Array.from(names));
 
   const output = new Array(r1 - r0 + 1);
   for (let r = r0; r <= r1; r++) {
-    const row = (output[r - r0] = Object.defineProperty({}, "#", {
+    // Should we be using Object.create(null) instead of an empty object here?
+    const row = (output[r - r0] = Object.defineProperty({}, "#", { // what is this non-enumerable row["#"] property for?
```

Comment:
For aid in range specification when using with Inputs.table in a notebook. The "#" column is included in the list of column names, so it shows up for every row. And even if Inputs.table gets a row number display in the future, after using the `range` option to skip an initial set of rows, it could never know what the source row number would have been.

<img width="180" alt="132925195-8ede0efb-e6b5-4cc9-ac8b-ca5ba9a27774" src="https://user-images.githubusercontent.com/10444/133497706-82c43ea9-ed8e-42ce-a18a-cf6df708c303.png">